apps/zbugs/shared/validators.ts

```diff
@@ -0,0 +1,70 @@
+import type {Query, Transaction} from '@rocicorp/zero';
+import {jwtVerify, type JWK, type JWTPayload} from 'jose';
+import type {schema} from './schema.ts';
+import {must} from '../../../packages/shared/src/must.ts';
+import {assert} from '../../../packages/shared/src/asserts.ts';
+
+const publicJwk = JSON.parse(
+  process?.env?.VITE_PUBLIC_JWK ?? import.meta.env.VITE_PUBLIC_JWK,
+) as JWK;
+
+export async function verifyToken(
```

Comment:
Few things here:

- kinda weird to re-parse this constantly. can we find a nice pattern to not do that?
- in case where token is *not* encrypted, should we skip verification? this prevents a little complexity of having to get the public key to the client.
- as discussed i think we should get rid of `userID` to simplify the model. I guess technically we should deprecate it to avoid a breaking change for now. Kind of a different task but maybe you could do as long as you're in there.